lib/api/controllers/authController.js

```diff
@@ -105,7 +109,40 @@ class AuthController {
    * @returns {Promise<object>}
    */
   getCurrentUser(request) {
-    return formatProcessing.formatUserForSerialization(this.kuzzle, request.context.user);
+    let formattedUser;
+
+    return formatProcessing.formatUserForSerialization(this.kuzzle, request.context.user)
+      .then(result => {
+        const
+          availableStrategies = this.kuzzle.pluginsManager.listStrategies(),
+          promises = [];
+
+        formattedUser = result;
+        
+        if (this.kuzzle.repositories.user.anonymous()._id === request.context.token.userId) {
+          return Promise.resolve([]);
+        }
+
+        for (let i = 0; i < availableStrategies.length; i++) {
```

Comment:
Already discussed with David: another drawback of using `Array.forEach` (or other array iteration methods) is that it doesn't iterate generator functions. I thought it would :sweat: 

We already agreed that using `for...of` is the cleanest way to iterate over iterables.